blogs/posts/2024-01-17_nearest_neighbour.qmd

```diff
@@ -0,0 +1,194 @@
+
+```
+
+To provide a suitable estimate of need for the newer practices without values, all the practices in the dataset were geocoded[^2] using the `geocode` function from the {tidygeocoder} package.
+
+[^2]: Geocoding is the process of converting addresses (often the postcode) into geographic coordinates (such as latitude and longitude) that can be plotted on a map.
+
+```{r}
+#| output: false
+practices <- practices |>
+  mutate(id = row_number()) |>
+  geocode(postalcode = postcode) |>
+  st_as_sf(coords = c("long", "lat"), crs = 4326)
+
+
+```
+
+```{r}
+#| code-fold: true
+#| warning: false
+#| message: false
+practices |>
+  gt()
+```
+
+This map shows the practices, purple are the practices with no need data and the yellow are practices with need data available.
+
+```{r}
+#| code-fold: true
+#| warning: false
+#| message: false
+# make map to display practices
+leaflet(practices) |> 
+  addTiles() |>
+  addCircleMarkers(color = ~pal(has_orig_need)) 
+
+```
+
+The data was split into those with, and without, a value for need. Using `st_join` from the {sf} package to join those without, and those with, a value for need, using the geometry to find all those within 1500m (1.5km).
+
+```{r}
+
+no_need <- practices |>
+  filter(has_orig_need == 0)
+
+with_need <- practices |>
+  filter(has_orig_need == 1)
+
+
+neighbours <- no_need |>
+  select(no_need_postcode = postcode,no_need_prac_code=practice_code) |>
+  st_join(with_need, st_is_within_distance, 1500) |>
+  st_drop_geometry() |>
+  select(id, no_need_postcode,no_need_prac_code) |>
+  inner_join(x = with_need, by = join_by("id")) 
+
+
+```
+
+```{r}
+#| code-fold: true
+#| warning: false
+#| message: false
+
+
+leaflet(neighbours) |> 
+  addTiles() |>
+  addCircleMarkers(color = "purple") |>
+  addMarkers(  -1.50686326666667, 52.4141089666667, popup = "Practice with no data"
+) |>
+  addCircles(-1.50686326666667, 52.4141089666667,radius=1500) |>
+  addMarkers(-1.46927, 52.51899, popup = "Practice with no data"
+) |>
+addCircles(-1.46927, 52.51899,radius=1500)
+```
+
+The data for the "neighbours" was grouped by the practice code of those without need data and a mean value calculated for each practice to generate an estimated value.
+
+```{r}
+
+neighbours_estimate <- neighbours |>
+  group_by(no_need_prac_code) |>
+    summarise(need_est=mean(value)) |>
+    st_drop_geometry(select(no_need_prac_code,need_est)) 
+
+
+
+```
+
+The original data was joined back to the "neighbours".
+
+```{r}
+
+  practices_with_neighbours_estimate <- practices |>
+    left_join(neighbours_estimate, join_by(practice_code==no_need_prac_code))  |>
+    st_drop_geometry(select(practice_code,need_est))
+```
+
+```{r}
+#| code-fold: true
+#| warning: false
+#| message: false
+  practices_with_neighbours_estimate |>
+    select(-has_orig_need,-id) |>
+    gt()
+
+```
+
+Finally, an updated dataframe was created of the need data using the actual need for the practice where available, otherwise using estimated need.
```

Comment:
data frame
```suggestion
Finally, an updated data frame was created of the need data using the actual need for the practice where available, otherwise using estimated need.
```